src/openai/_client.py

```diff
@@ -122,11 +125,12 @@ def __init__(
         """
         if api_key is None:
             api_key = os.environ.get("OPENAI_API_KEY")
-        if api_key is None:
+        if api_key is None and bearer_token_provider is None:
             raise OpenAIError(
                 "The api_key client option must be set either by passing api_key to the client or by setting the OPENAI_API_KEY environment variable"
```

Comment:
Should we update the error message to include `bearer_token_provider`?

Reply:
Also, should we error if both are provided?

Reply:
Done.